src/main.rs

```diff
@@ -0,0 +1,129 @@
+                                .conflicts_with("patch")
+                                .conflicts_with("minor")
+                                .help("increase major version (sets minor and patch versions to 0)"))
+                          .arg(Arg::with_name("minor")
+                                .short("m")
+                                .long("minor")
+                                .multiple(false)
+                                .conflicts_with("patch")
+                                .help("increase minor version (sets patch version to 0)"))
+                          .arg(Arg::with_name("patch")
+                                .short("p")
+                                .long("patch")
+                                .multiple(false) 
+                                .help("increase patch version"))
+                          .arg(Arg::with_name("pre")
+                                .long("pre")
+                                .takes_value(true)
+                                .multiple(true) 
+                                .help("set pre-release version (alpha, beta, ...)"))
+                          .arg(Arg::with_name("meta")
+                                .long("meta")
+                                .takes_value(true)
+                                .multiple(true)
+                                .help("set metadata of version (ci, commit, ...)"))
+                          .get_matches();
+    
+    // Since VERSION is required(true), unwrap is safe.
+    let input_version = matches.value_of("VERSION").unwrap();
+
+    // Try to parse input version (validate correctness), if error print it and return non-zero status code.
+    let mut version = match Version::parse(input_version) {
+          Ok(version) => version,
+          Err(e) => {
+                println!("{}", e);
+                return 1;
+          }
+    };
+
+    // Increase major version.
+    if matches.is_present("major") {
+        version.increment_major();
+    }
+
+    // Increase minor version.
+    if matches.is_present("minor") {
+        version.increment_minor();
+    }
+
+    // Increase patch version.
+    if matches.is_present("patch") {
+        version.increment_patch();
+    }
+
+    // Modify pre-release.
+    if matches.is_present("pre") {
+        for value in matches.values_of("pre").unwrap() {
+            push_identifier(&mut version.pre, value);
+        }
+    }
+
+    // Modify metadata.
+    if matches.is_present("meta") {
+        for value in matches.values_of("meta").unwrap() {
+            push_identifier(&mut version.build, value);
+        }
+    }
+
+    println!("{}", version.to_string());
+    0
+}
+
+// Validation for empty strings.
+fn is_not_empty(val: String) -> Result<(), String> {
+    if !val.is_empty() {
+        Ok(())
+    } else {
+        Err(String::from("VERSION must not be empty"))
+    }
+}
+
+// Borrowed from semver-parser (common is not public).
```

Comment:
Well, making common public would help, I was planning to make a PR on that, after you took a look at this (I'm not that proficient in rust). Also, since I'm using regex here, but it is used in the chain - semver -> semver-parser -> regex, does it make a difference for the user? They have to get it anyways...